pkg/process/checks/process.go

```diff
@@ -140,17 +144,11 @@ func (p *ProcessCheck) Init(syscfg *SysProbeConfig, info *HostInfo, oneShot bool
 
 	p.notInitializedLogLimit = log.NewLogLimit(1, time.Minute*10)
 
-	var tu net.SysProbeUtil
-
-	if syscfg.NetworkTracerModuleEnabled {
-		// Calling the remote tracer will cause it to initialize and check connectivity
-		tu, err = net.GetRemoteSystemProbeUtil(syscfg.SystemProbeAddress)
-		if err != nil {
-			log.Warnf("could not initiate connection with system probe: %s", err)
-		}
+	if syscfg.NetworkTracerModuleEnabled || syscfg.ProcessModuleEnabled {
+		p.sysprobeClient = client.Get(syscfg.SystemProbeAddress)
```

Comment:
Could you explain why there is now a check for `syscfg.ProcessModuleEnabled` - it's not obvious why this is needed

Reply:
It was part of `getRemoteSysProbeUtil` before: https://github.com/DataDog/datadog-agent/pull/31365/files/b909fbb7526718c8813cbe7fa1b658d70e03517b#diff-cf987be7da12570b59c9ab8cb5ff3cfd3743063e93fb936fd6e51fae028d7aceL664-L667

Reply:
Right, what wasn't obvious was why this check needs to be here now, whereas before we only checked `syscfg.NetworkTracerModuleEnabled`. Looks like it's because we're now caching the system probe client for later use